snetd/cmd/flags.go

```diff
@@ -0,0 +1,48 @@
+package cmd
+
+import (
+	"github.com/singnet/snet-daemon/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cfgFile            = ServeCmd.PersistentFlags().StringP("config", "c", "snetd.config", "config file")
+	daemonType         = ServeCmd.PersistentFlags().StringP("type", "t", "grpc", "daemon type: one of 'grpc','http'")
+	blockchainEnabled  = ServeCmd.PersistentFlags().BoolP("blockchain", "b", true, "enable blockchain processing")
+	listenPort         = ServeCmd.PersistentFlags().IntP("port", "p", 5000, "daemon listen port")
+	ethEndpoint        = ServeCmd.PersistentFlags().String("ethereum-endpoint", "http://127.0.0.1:8545", "ethereum JSON-RPC endpoint")
+	mnemonic           = ServeCmd.PersistentFlags().String("mnemonic", "", "HD wallet mnemonic")
+	hdwIndex           = ServeCmd.PersistentFlags().Int("wallet-index", 0, "HD wallet index")
+	dbPath             = ServeCmd.PersistentFlags().String("db-path", "snetd.db", "database file path")
+	passthroughEnabled = ServeCmd.PersistentFlags().Bool("passthrough", false, "passthrough mode")
+	serviceType        = ServeCmd.PersistentFlags().String("service-type", "grpc", "service type: one of 'grpc','jsonrpc','process'")
+	wireEncoding       = ServeCmd.PersistentFlags().String("wire-encoding", "proto", "message encoding: one of 'proto','json'")
+	pollSleep          = ServeCmd.PersistentFlags().String("poll-sleep", "5s", "blockchain poll sleep time")
+)
+
+func init() {
+	rf := ServeCmd.PersistentFlags()
+	vip := config.Vip()
+
+	vip.BindPFlag(config.ConfigPathKey, rf.Lookup("config"))
+	vip.BindPFlag(config.DaemonTypeKey, rf.Lookup("type"))
+	vip.BindPFlag(config.BlockchainEnabledKey, rf.Lookup("blockchain"))
+	vip.BindPFlag(config.DaemonListeningPortKey, rf.Lookup("port"))
+	vip.BindPFlag(config.EthereumJsonRpcEndpointKey, rf.Lookup("ethereum-endpoint"))
+	vip.BindPFlag(config.HdwalletMnemonicKey, rf.Lookup("mnemonic"))
+	vip.BindPFlag(config.HdwalletIndexKey, rf.Lookup("wallet-index"))
+	vip.BindPFlag(config.DbPathKey, rf.Lookup("db-path"))
+	vip.BindPFlag(config.PassthroughEnabledKey, rf.Lookup("passthrough"))
+	vip.BindPFlag(config.ServiceTypeKey, rf.Lookup("service-type"))
+	vip.BindPFlag(config.WireEncodingKey, rf.Lookup("wire-encoding"))
+	vip.BindPFlag(config.PollSleepSecsKey, rf.Lookup("poll-sleep"))
+
+	cobra.OnInitialize(func() {
+		vip.SetConfigName(*cfgFile)
```

Comment:
Does this need to be here and in config.go?

Reply:
it functions correctly as-is, in that the `config/` package can be used without any command-line (or cobra) code and it will read from the location specified by the environment variable.

if the intention is that config is only used by the serving code, we can remove the call to "setconfigname" in config.go and only set it once here.

Reply:
My vote is remove it from config.go -- I don't think we need to maintain functionality in anything beyond snetd command currently

If someone were to use config.go outside the context of cobra, doesn't this PR remove all defaults that were previously set within viper (but are now set within cobra?)